File: Examples/Objective-C/Boom/PhoneViewController.m
//
//  ViewController.m
//  Boom
//
//  Created by Sergey P on 31.05.2018.
//  Copyright © 2018 Boomware. All rights reserved.
//

#import "PhoneViewController.h"
#import <Boomware/Boomware.h>

@interface PhoneViewController ()

@property (weak, nonatomic) IBOutlet UITextField *phoneField;
@property (weak, nonatomic) IBOutlet UISegmentedControl *segmentedControl;

@end

@implementation PhoneViewController

#pragma mark - Lifecycle

- (void)viewDidLoad
{
    [super viewDidLoad];

    self.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:NSLocalizedString(@"Remvoe device data", nil) style:UIBarButtonItemStylePlain target:self action:@selector(actionRemoveDeviceData:)];
}

#pragma mark - Actions

- (IBAction)actionGo:(id)sender
{
    __weak __typeof(self) weakSelf = self;
    [BWBoomware verifyNumber:self.phoneField.text method:[self selectedVerificationMethod] completion:^(NSError * _Nullable error) {
       
        if (! error)
        {
            [self performSegueWithIdentifier:@"PINSegue" sender:sender];
        }
        else
        {
            UIAlertController *alert = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"Error", nil) message:error.localizedDescription preferredStyle:UIAlertControllerStyleAlert];
            UIAlertAction *action = [UIAlertAction actionWithTitle:NSLocalizedString(@"Cancel", nil) style:UIAlertActionStyleCancel handler:nil];
            [alert addAction:action];
            [weakSelf.navigationController presentViewController:alert animated:YES completion:nil];
        }
        
    }];
}

- (IBAction)actionRemoveDeviceData:(id)sender
{
    __weak __typeof(self) weakSelf = self;
    [BWBoomware removeDataWithCompletion:^(NSError * _Nullable error) {

        if (error)
        {
            UIAlertController *alert = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"Error", nil) message:error.localizedDescription preferredStyle:UIAlertControllerStyleAlert];
            UIAlertAction *action = [UIAlertAction actionWithTitle:NSLocalizedString(@"Cancel", nil) style:UIAlertActionStyleCancel handler:nil];
            [alert addAction:action];
            [weakSelf.navigationController presentViewController:alert animated:YES completion:nil];
        }
        else
        {
            UIAlertController *alert = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"Success", nil) message:NSLocalizedString(@"Device removed", nil) preferredStyle:UIAlertControllerStyleAlert];
            UIAlertAction *action = [UIAlertAction actionWithTitle:NSLocalizedString(@"OK", nil) style:UIAlertActionStyleCancel handler:nil];
            [alert addAction:action];
            [weakSelf.navigationController presentViewController:alert animated:YES completion:nil];
        }
    }];
}

#pragma mark - Private

- (BWBoomwareMethod)selectedVerificationMethod
{
    switch (self.segmentedControl.selectedSegmentIndex)
    {
        case 0:
            return BWBoomwareMethodSMS;
            
        case 1:
            return BWBoomwareMethodVoice;
            
        default:
            return BWBoomwareMethodCall;
    }
}

@end

File: Examples/Swift/Boom/PhoneViewController.swift
//
//  ViewController.swift
//  Boom
//
//  Created by Sergey P on 31.05.2018.
//  Copyright © 2018 Boomware. All rights reserved.
//

import UIKit
import Boomware

class PhoneViewController: UIViewController
{

    @IBOutlet weak var phoneField: UITextField!
    @IBOutlet weak var segmentedControl: UISegmentedControl!
    
    // MARK: Actions
    
    @IBAction func actionGo(_ sender: Any) {
        guard let number = self.phoneField.text else {
            return
        }

        guard let params = BWParams(method: self.selectedVerificationMethod(), phoneNumber: number) else {
            return
        }

        params.from = nil

        BWBoomware.verify(params: params) { (error) in
            if let err = error as NSError? {
                let alert = UIAlertController(title: "Error", message: err.localizedDescription, preferredStyle: .alert)
                let action = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
                alert.addAction(action)
                self.present(alert, animated: true, completion: nil)
            } else {

                self.performSegue(withIdentifier: "PINSegue", sender: nil)
            }
        }
    }
    
    // MARK: Private
    private func selectedVerificationMethod() -> String
    {
        switch segmentedControl.selectedSegmentIndex
        {
        case 0:
            return BWParamsMethodSMS
            
        case 1:
            return BWParamsMethodVoice
            
        default:
            return BWParamsMethodCall
        }
    }

}

